### PR TITLE
Edit event schema

### LINKED ISSUE #126: Revise event schema and add an array of objects for recurring events
# Intro
This issue has been opened specifically to give 100Devs members a chance to get involved. The issues presented will have more instruction and be easier to solve. (If you are already contributing to this project or your skills are well ahead of this issue, consider letting someone else take this on!)

## Summary
Together needs a way to store the recurring events in the database after they are generated. 

You will need to make some changes to the existing model schema to accommodate implementing recurring dates.

This will involve using an array in the MongoDB document to embed objects for each event as well as combining the existing start/end dates and times into two distinct Date objects. 

## Tasks
You will only need to edit `/server/models/Event.js` 
* `startDate` and `startTime` will be combined into a single Date type 
* `endDate` and `endTime` will be combined into a single Date type
* `recurringRate` will be renamed to `recurring` and be a required boolean type 
* `recurringDates` array will be used to embed recurring events
    * each object in the array represents a unique editable event and will be used to contain its own copy of the event's title, description, start date/time, end date/time, location, and ~~Discord username~~.
    * `recurringDates` will be `require: false`

### Steps 

A feature branch named `feature/modify-event-schema` has been created for this specific issue. The [contributing guidelines](https://github.com/Caleb-Cohen/Together/blob/development/.github/CONTRIBUTING.md) contain the detailed walkthrough of the steps listed below.

1. Fork the development branch, clone it to your device, and check out the `feature/modify-event-schema` branch.
1. Create and check out an additional branch with a descriptive name on your device.
1. Edit the `/server/models/Event.js` file with the appropriate changes
1. Commit your work. 
1. Push your changes to GitHub.
1. Submit a pull request from GitHub.

## Potential Issues
Some minor alterations or additional changes may be requested during testing.

## Additional Notes and reference links
🚨 [6 Rules of thumb for MongoDB schema design.](https://www.mongodb.com/blog/post/6-rules-of-thumb-for-mongodb-schema-design)
🚨 [MongoDB schema design best practices.](https://www.mongodb.com/developer/products/mongodb/mongodb-schema-design-best-practices/)

---
**When completed please ensure you are creating a pull request into the development branch, instead of main.**

**Please comment below if you'd like to take this issue on.**  👇

### PR DESCRIPTION
**Pull request by intelagense**
# Description

Edited the model event schema to prepare for recurring date functionality. 

## Type of change

Please select everything applicable. Please, do not delete any lines.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Issue
- [x] Is this related to a specific issue? Is so please specify:
Closes #126

# Checklist:

- [x] I have executed `npm run lint` and resolved any outstanding errors. Most issues can be solved by executing `npm run format`
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by intelagense** [comment]
@MatthewBozin, @JoeDravarol 👀 Can you review or comment?

**Comment by Caleb-Cohen** [comment]
@intelagense I've unchecked "Generates no new warnings" because this is a breaking change and forms cannot submit.

Do we have someone working on the conforming the form to match the updated schema? All that work should be done within the `modify-event-schema` branch.

**Comment by MatthewBozin** [comment]
Are the "recurringDates" properties for "title", "description", and "location" expected to vary from those of the event object? 

If not, maybe we can get rid of those values within "recurringDates" and have the code that generates those events pull those properties from the event object. Our data in mongoDB would be much simpler that way!

**Comment by MatthewBozin** [comment]
@Caleb-Cohen how urgently do we need the form fix? Since it's fairly limited in scope, if it's not so urgent we could make it a good-100devs-first-issue!

**Comment by Caleb-Cohen** [comment]
> @Caleb-Cohen how urgently do we need the form fixed? Since it's fairly limited in scope, if it's not so urgent we could make it a good-100devs-first-issue!

I agree. I'll get it posted. Working on getting the priority tiers done for the open tasks.

**Comment by intelagense** [comment]
> Are the "recurringDates" properties for "title", "description", and "location" expected to vary from those of the event object?
> 
> If not, maybe we can get rid of those values within "recurringDates" and have the code that generates those events pull those properties from the event object. Our data in mongoDB would be much simpler that way!

@MatthewBozin those fields won't vary for MVP release but I wanted to put something in place so we wouldn't have to change the schema when we add the ability to edit events. 🤞 My fingers are crossed that it will actually help us in the future.

File: server/models/Event.js [hunk 1]
```diff
@@ -9,29 +9,27 @@ const EventSchema = new mongoose.Schema({
...
-  },
-  endTime: {
-    type: String,
-    required: true,
-  },
+  recurringDates: [
+    {
+      title: { type: String },
+      description: { type: String },
+      start: { type: Date },
```
Comment (Caleb-Cohen) [inline server/models/Event.js:28]: this will be the entire ISO 8601 start date?

File: server/models/Event.js [hunk 2]
```diff
@@ -9,29 +9,27 @@ const EventSchema = new mongoose.Schema({
...
     require: true,
   },
-  recurringDates: {
-    type: Array,
-  },
-  endTime: {
-    type: String,
-    required: true,
-  },
+  recurringDates: [
```
Comment (Caleb-Cohen) [inline server/models/Event.js:24]: I agree with putting title, description, start, end in the recurring dates array. I still think title and description should also be stored outside the recurring dates array.

**Review by Caleb-Cohen — COMMENTED** [review]
I left some comments specifically regarding recurringDates. I also have some concerns with using "start" and "end" from a clarity perspective.

**Pull request merged**